src/app/language-support/page.tsx

```diff
@@ -1,127 +1,104 @@
 'use client';
 
-import { useEffect, useState } from 'react';
+import { useCallback, useEffect, useMemo, useState } from 'react';
 import { getAllCases } from '@/api/supabase/queries/cases';
 import { getAllDocuments } from '@/api/supabase/queries/documentTranslation';
 import { getAllInterpretation } from '@/api/supabase/queries/interpretation';
-import { AllLanguageSupport } from '@/types/schema';
+import ListingPage from '@/components/ListingPage';
+import { LanguageSupport, Listing } from '@/types/schema';
 import { timestampStringToDate } from '@/utils/helpers';
 
+const typeOptions = new Map([
+  ['DOC', 'Document Translation'],
+  ['INT', 'One-time Interpretation'],
+  ['CASE', 'Case Interpretation'],
+]);
+
 export default function Page() {
-  const [allLanguageSupport, setAllLanguageSupport] = useState<
-    AllLanguageSupport[]
-  >([]);
+  const [lsData, setLSData] = useState<LanguageSupport[]>([]);
+  const [typeFilters, setTypeFilters] = useState(new Set<string>());
+  const [languageFilters, setLanguageFilters] = useState(new Set<string>());
+  const [selectedListing, setSelectedListing] = useState<Listing | null>(null);
 
+  // load language support on first render
   useEffect(() => {
-    const fetchData = async () => {
+    (async () => {
       try {
         const [docListings, intListings, casesInterpretationListings] =
           await Promise.all([
             getAllDocuments(),
             getAllInterpretation(),
             getAllCases(),
           ]);
+
         const sortedLS = [...docListings, ...intListings].sort(
           (a, b) =>
             timestampStringToDate(b.upload_date).getTime() -
             timestampStringToDate(a.upload_date).getTime(),
         );
-        setAllLanguageSupport([
+
+        const lsDataToSet = [
           ...casesInterpretationListings.filter(
             caseInterpretation => caseInterpretation.needs_interpreter === true,
           ),
           ...sortedLS,
-        ]);
+        ];
+
+        setLSData(lsDataToSet);
+        setSelectedListing(lsDataToSet[0]);
```

Comment:
nit: do we want to do some error handling in case `lsDataToSet` is empty (though that case will probably be rare)?

Reply:
nice catch! yep, I'll add a conditional